File: fsharp/nth-prime/NthPrime.fs
module NthPrime

let prime nth : int option = 
    let rec finder primes current = 
        if List.length primes >= nth then List.tryLast primes
        elif List.exists (fun n -> current % n = 0) primes then finder primes (current + 1)
        else finder (primes @ [current]) (current + 1)
    finder [] 2

File: go/nth-prime/nth_prime.go
package prime

func Nth(n int) (int, bool) {
	if n < 1 {
		return 0, false
	}

	soFar := []int{2}
	c := 3
	for {
		if len(soFar) == n {
			return soFar[len(soFar)-1], true
		}

		isPrime := true
		for _, p := range soFar {
			if c%p == 0 {
				isPrime = false
				break
			}
		}

		if isPrime {
			soFar = append(soFar, c)
		}
		c++
	}
}
